src/test/scala/PropertyCheckerTest.scala

```diff
@@ -0,0 +1,464 @@
+  }
+
+  test("numeric property returns invalid on negative values") {
+    val (_, warnings, _) = PropertyChecker.checkProperty("headerRowCount", new IntNode(-10), "", "und")
+    assert(warnings === Array[String]("invalid_value"))
+  }
+
+  test("numeric property returns value without warnings on valid value") {
+    val (value, warnings, _) = PropertyChecker.checkProperty("headerRowCount", new IntNode(5), "", "und")
+    assert(warnings === Array[String]())
+    assert(value === new IntNode(5))
+  }
+
+  test("null property returns value in array without warnings on valid value") {
+    val (values, warnings, _) = PropertyChecker.checkProperty("null", new TextNode("sample value"), "", "und")
+    var expectedValues = new TextNode("sample value")
+    assert(warnings === Array[String]())
+    assert(values === expectedValues)
+  }
+
+  test("null property returns warnings for invalid value (non string type)") {
+    val (_, warnings, _) = PropertyChecker.checkProperty("null", BooleanNode.getFalse, "", "und")
+    assert(warnings === Array[String]("invalid_value"))
+  }
+
+  test("null property with values array holding different types") {
+    val values = JsonNodeFactory.instance.arrayNode()
+    values.add("sample")
+    values.add(5)
+    val expectedValues = JsonNodeFactory.instance.arrayNode()
+    expectedValues.add("sample")
+
+    val (returnedValues, warnings, _) = PropertyChecker.checkProperty("null", values, "", "und")
+
+    assert(warnings === Array[String]("invalid_value"))
+    assert(returnedValues === expectedValues)
+  }
+
+  test("separator property returns invalid warning if not of type string or null") {
+    val (_, warnings, _) = PropertyChecker.checkProperty("separator", BooleanNode.getFalse, "", "und")
+    assert(warnings === Array[String]("invalid_value"))
+  }
+
+  test("separator property returns value on valid case") {
+    val (returnedValue, _, _) = PropertyChecker.checkProperty("separator", JsonNodeFactory.instance.objectNode(), "", "und")
+    assert(returnedValue === NullNode.getInstance())
+  }
+
+  test("datatype property checker returns expected value for object with correct base and format") {
+    val json = "{\"base\": \"string\", \"format\": \"^The Sample RegEx$\"}"
+    val jsonNode = objectMapper.readTree(json)
+    val expectedJsonNode = JsonNodeFactory.instance.objectNode()
+    expectedJsonNode.put("base", "http://www.w3.org/2001/XMLSchema#string")
+    expectedJsonNode.put("format", "^The Sample RegEx$")
+
+    val (returnedValue, warnings, _) = PropertyChecker.checkProperty("datatype", jsonNode, "", "und")
+
+    assert(returnedValue === expectedJsonNode)
+    assert(warnings === Array[String]())
+  }
+
+  val t = Tag("Some Group")
+  test("datatype property checker returns expected value for object with invalid base and format", t) {
+    val json =
+      """
+        |{
+        | "base": "invalidDatatypeSupplied",
+        | "format": "^The Sample RegEx$"
```

Comment:
Do the CSVW format regexes allow you to do partial matches or do they have to be whole string matches? That might be something we need to look into.

Reply:
Not sure on this. This part is not fully complete as we haven't implemented the NumberFormat and DateFormat classes. The RegEx in format is mainly consumed in these classes.

Reply:
Okay, it'll be more important when we get to validating the CSV string column values. We'll leave it until then.